components/home/Challenge.tsx

```diff
@@ -56,7 +56,7 @@ export function Challenge({
         setIsLiked(likes.includes(currentUser.uid));
         setLikes(likes);
       });
-  }, [challengeDB, firestoreCtrl, likes]);
+  }, [challengeDB.challenge_id]);
```

Comment:
Why not fetch the likes each time the array "likes" does change ? It could be another user updating it, it would be nice to have it displayed directly.

Reply:
that is not how it works. Or at least how the useEffect is created here. Firestore is already provided in the challenge itself, and likes isn't used as there is already there in the lamba function